spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -17,3 +17,28 @@ def foobar_action
     end
   end
 end
+
+RSpec.describe Hq::PartnersController, type: :controller do
+  before :each do
+    sign_in instance_double(AdminUser)
+    @partner = build_stubbed :partner
+    expect(Partner).to receive(:find).and_return(@partner)
+  end
+
+  it 'editing renders the edit view' do
+    get :edit, id: @partner.id
+    expect(response).to render_template 'edit'
+  end
+
+  it 'unsuccessful update renders the edit view' do
+    expect(@partner).to receive(:save).and_return(false)
+    patch :update, id: @partner.id, partner: {name: ""}
```

Comment:
Rather out of curiosity - does `.and_return(false)` above make `partner: { name: "" }` unnecessary?